finality-aleph/src/new_network/data_network/split.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+    new_network::data_network::{
+        aleph_network::{AlephNetwork, AlephNetworkData},
+        rmc_network::{RmcNetwork, RmcNetworkData},
+        Recipient, SessionCommand,
+    },
+    Error, NodeIndex, SessionId,
+};
+use codec::{Codec, Decode, Encode};
+use futures::{channel::mpsc, Future, FutureExt, StreamExt};
+use log::{debug, trace, warn};
+use sp_api::BlockT;
+
+#[derive(Clone, Encode, Decode, Debug)]
+pub(crate) enum NetworkData<B: BlockT> {
+    Aleph(AlephNetworkData<B>),
+    Rmc(RmcNetworkData<B>),
+}
+
+pub(crate) struct DataNetwork<D: Clone + Codec> {
+    session_id: SessionId,
+    data_from_consensus_network: mpsc::UnboundedReceiver<D>,
+    commands_for_consensus_network: mpsc::UnboundedSender<SessionCommand<D>>,
+}
+
+impl<D: Clone + Codec> DataNetwork<D> {
+    fn new(
+        session_id: SessionId,
+        data_from_consensus_network: mpsc::UnboundedReceiver<D>,
+        commands_for_consensus_network: mpsc::UnboundedSender<SessionCommand<D>>,
+    ) -> Self {
+        DataNetwork {
+            session_id,
+            data_from_consensus_network,
+            commands_for_consensus_network,
+        }
+    }
+
+    pub(crate) fn send(&self, data: D, recipient: Recipient<NodeIndex>) -> Result<(), Error> {
+        let sc = SessionCommand::Data(self.session_id, data, recipient);
+        self.commands_for_consensus_network
+            .unbounded_send(sc)
+            .map_err(|_| Error::SendData)
+    }
+
+    pub(crate) async fn next(&mut self) -> Option<D> {
+        self.data_from_consensus_network.next().await
+    }
+}
+
+pub(crate) fn split_network<B: BlockT>(
+    data_network: DataNetwork<NetworkData<B>>,
+    data_store_tx: mpsc::UnboundedSender<AlephNetworkData<B>>,
```

Comment:
Ok, yeah, in this form `AlephNetwork` and `RmcNetwork` are quite entangled in everything here, ignore the comment about moving them to a separate module. :/

Reply:
Ignored above